File: psra/model/src/main/scala/by/arhor/psra/repository/BasicRepository.scala
package by.arhor.psra.repository

import java.util.Optional

import org.springframework.data.domain.{Page, Pageable}

trait BasicRepository[T, K] {

  def findByIdAndEnabledTrue(id: K): Optional[T]

  def findAllByEnabledTrue(pageable: Pageable): Page[T]

}

File: psra/model/src/main/java/by/arhor/psra/repository/BasicRepository.java
package by.arhor.psra.repository;

import java.util.Optional;

import org.springframework.data.domain.Pageable;
import org.springframework.data.domain.Page;

public interface BasicRepository<T, K> {

  Optional<T> findByIdAndEnabledTrue(K id);

  Page<T> findAllByEnabledTrue(Pageable pageable);

}
